components/brave_wallet/common/f4_jumble_unittest.cc

```diff
@@ -0,0 +1,261 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/f4_jumble.h"
+
+#include <string>
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(F4JumbleUnitTest, H_Personalizer) {
+  std::string personalizer = "UA_F4Jumble_H";
+  personalizer.push_back('\x07');
+  personalizer.push_back(0);
+  personalizer.push_back(0);
+  ASSERT_EQ(GetHPersonalizer(7), personalizer);
+}
+
+TEST(F4JumbleUnitTest, G_Personalizer) {
+  {
+    std::string personalizer = "UA_F4Jumble_G";
+    personalizer.push_back('\x07');
+    personalizer.push_back('\x0d');
+    personalizer.push_back('\x00');
+    ASSERT_EQ(GetGPersonalizer(7, 13), personalizer);
+  }
+
+  {
+    std::string personalizer = "UA_F4Jumble_G";
+    personalizer.push_back('\07');
+    personalizer.push_back('\xff');
+    personalizer.push_back('\xff');
+    ASSERT_EQ(GetGPersonalizer(7, 65535), personalizer);
+  }
+}
+
+// https://github.com/zcash/librustzcash/blob/f4jumble-0.1.0/components/f4jumble/src/test_vectors.rs
+TEST(F4JumbleUnitTest, ApplyF4Jumble) {
+  {
+    std::vector<uint8_t> input = {};
+
+    std::vector<uint8_t> expected = {};
+
+    EXPECT_EQ(expected, ApplyF4Jumble(input));
+  }
+
+  {
```

Comment:
We need to test Apply + Revert resulting in original input. Also covering corner cases for input sizes.
Something like 
```
std::vector<uint8_t> input(1000, 'a');
auto input_span = make_span(input);
for (int i = 0; i < input_span.size(); ++i) {
  auto d = input_span.subspan(0, i);
  EXPECT_EQ(d, RevertF4Jumble(ApplyF4Jumble(d)));
}
```

Reply:
Done